### PR TITLE
[Docs] Translate the config tutorial

### PR DESCRIPTION
**Pull request by HAOCHENYE**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docs/en/advanced_tutorials/config.md [hunk 1]
```diff
@@ -1,3 +1,579 @@
 # Config
 
-Coming soon. Please refer to [chinese documentation](https://mmengine.readthedocs.io/zh_CN/latest/tutorials/config.html).
+MMEngine implements an abstract configuration class (`Config`) to provide a unified configuration access interface for users. `Config` supports different type of configuration file, including `python`, `json` and `yaml`, and you can choose the type according to your preference. `Config` override some magic method, this could help you access the data stored in `Config` just like getting value from `dict`, or getting attribute from instance. Besides, `Config` also provides an inheritance mechanism, it could help us better organize and manage the configuration files.
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/config.md:3]:
```suggestion
MMEngine implements an abstract configuration class (`Config`) to provide a unified configuration access interface for users. `Config` supports different type of configuration file, including `python`, `json` and `yaml`, and you can choose the type according to your preference. `Config` overrides some magic method, which could help you access the data stored in `Config` just like getting values from `dict`, or getting attributes from instances. Besides, `Config` also provides an inheritance mechanism, which could help you better organize and manage the configuration files.
```

File: docs/en/advanced_tutorials/config.md [hunk 2]
```diff
@@ -1,3 +1,579 @@
...
+print(cfg)
+```
+
+```
+Config (path: learn_read_config.py): {'test_int': 1, 'test_list': [1, 2, 3], 'test_dict': {'key1': 'value1', 'key2': 0.1}}
+```
+
+## How to use `Config`
+
+After loading the configuration file, we can access the data stored in `Config` instance just like getting/writing value from `dict`, or getting/writing attribute from instance.
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/config.md:77]:
```suggestion
After loading the configuration file, we can access the data stored in `Config` instance just like getting/setting values from `dict`, or getting/setting attributes from instances.
```

File: docs/en/advanced_tutorials/config.md [hunk 3]
```diff
@@ -1,3 +1,579 @@
...
+[1, 2, 3]
+{'key1': 'value1', 'key2': 0.1}
+2
+[1, 2, 3]
+{'key1': 'value1', 'key2': 0.1}
+[1, 3, 3]
+```
+
+```{note}
+The `dict` object parsed by `Config` will be converted to `ConfigDict`, and then we can access the value of the `dict` the same as accessing the attribute of a instance.
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/config.md:103]:
```suggestion
The `dict` object parsed by `Config` will be converted to `ConfigDict`, and then we can access the value of the `dict` the same as accessing the attribute of an instance.
```

File: docs/en/advanced_tutorials/config.md [hunk 4]
```diff
@@ -1,3 +1,579 @@
...
+
+Here is an example of defining optimizers in a configuration file.
+
+`config_sgd.py`：
+
+```python
+optimizer = dict(type='SGD', lr=0.1, momentum=0.9, weight_decay=0.0001)
+```
+
+Suppose we have defined a registry `OPTIMIZERS`, which includes various optimizers, and then We can build the optimizer as below
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/config.md:116]:
```suggestion
Suppose we have defined a registry `OPTIMIZERS`, which includes various optimizers. Then we can build the optimizer as below
```

File: docs/en/advanced_tutorials/config.md [hunk 5]
```diff
@@ -1,3 +1,579 @@
...
+    nesterov: False
+    weight_decay: 0.0001
+)
+```
+
+## Inheritance between configuration files
+
+Sometimes, the difference between two different configuration files is so small that only one field may be changed. However, we need to copy and paste everything, and only modify one line. It is not easy to locate the specific difference when we go back to this after a long time.
+
+In another case, multiple configuration files may have the same batch of fields, and we have to copy and paste them in different configuration files. It will also be hard to maintain these fields in a long time.
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/config.md:149]:
```suggestion
In another case, multiple configuration files may have the same batch of fields, and we have to copy and paste them in different configuration files. It will also be hard to maintain these fields in a long time.

We address these issues with inheritance mechanism, detailed as below.
```

File: docs/en/advanced_tutorials/config.md [hunk 6]
```diff
@@ -1,3 +1,579 @@
...
+```
+
+`resnet50_runtime.py`：
+
+```python
+_base_ = ['optimizer_cfg.py', 'runtime_cfg.py']
+model = dict(type='ResNet', depth=50)
+```
+
+At this case, reading the `resnet50_runtime.py` will give you 3 fields `model`, `optimizer`, and `gpu_ids`.
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/config.md:194]:
```suggestion
In this case, reading the `resnet50_runtime.py` will give you 3 fields `model`, `optimizer`, and `gpu_ids`.
```

File: docs/en/advanced_tutorials/config.md [hunk 7]
```diff
@@ -1,3 +1,579 @@
...
+```python
+cfg = Config.fromfile('resnet50_runtime.py')
+print(cfg.optimizer)
+```
+
+```
+{'type': 'SGD', 'lr': 0.02, 'momentum': 0.9, 'weight_decay': 0.0001}
+```
+
+In this way, we can disassemble the configuration file, define some general configuration files, and inherit them in the specific configuration file. This could avoid to define a lot duplicated contents in multiple configuration files.
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/config.md:205]:
```suggestion
In this way, we can disassemble the configuration file, define some general configuration files, and inherit them in the specific configuration file. This could avoid defining a lot of duplicated contents in multiple configuration files.
```

File: docs/en/advanced_tutorials/config.md [hunk 8]
```diff
@@ -1,3 +1,579 @@
...
+```python
+cfg = Config.fromfile('resnet50_lr0.01.py')
+print(cfg.optimizer)
+```
+
+```
+{'type': 'SGD', 'lr': 0.01, 'momentum': 0.9, 'weight_decay': 0.0001}
+```
+
+For non-dictionary fields, such as integers, strings, lists, etc., they can be completely overwritten by redefining them, for example, the code block blow will change the value of the `gpu_ids` to `[0]`.
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/config.md:232]:
```suggestion
For non-dictionary fields, such as integers, strings, lists, etc., they can be completely overwritten by redefining them. For example, the code block below will change the value of the `gpu_ids` to `[0]`.
```

File: docs/en/advanced_tutorials/config.md [hunk 9]
```diff
@@ -1,3 +1,579 @@
...
+
+Although this way is general for all types of files, there are some syntactic limitations that prevent us from taking full advantage of the dynamic nature of the `python` configuration file. For example, if we want to modify a variable defined in `_base_`:
+
+```python
+_base_ = ['resnet50.py']
+a = {{_base_.model}}
+a['type'] = 'MobileNet'
+```
+
+The `Config` is not able to parse such a configuration file (it will raise an errors when parsing). The `Config` provides a more `pythonic` way to modify base variables for `python` configuration files.
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/config.md:295]:
```suggestion
The `Config` is not able to parse such a configuration file (it will raise an error when parsing). The `Config` provides a more `pythonic` way to modify base variables for `python` configuration files.
```

File: docs/en/advanced_tutorials/config.md [hunk 10]
```diff
@@ -1,3 +1,579 @@
...
+`dump_dict.py`
+
+```python
+a=1
+b=2
+```
+
+# Advanced uses
+
+In this section, we'll introduce some advanced uses of the `Config`, and some tips could make it easier for users to develop and use downstream repositories.
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/config.md:370]:
```suggestion
In this section, we'll introduce some advanced uses of the `Config`, and some tips that could make it easier for users to develop and use downstream repositories.
```

File: docs/en/advanced_tutorials/config.md [hunk 11]
```diff
@@ -1,3 +1,579 @@
...
+
+```{note}
+The standard procedure only supports modifying String, Integer, Floating Point, Boolean, None, List, and Tuple fields from the command line. For the elements of list and tuple instance, each of them must be one of the above seven types.
+```
+
+### import the custom module
+
+If we customize a module and register it into the corresponding registry, could we directly build it from the configuration file as the previous [section](#how-to-use-config) does? The answer is “I don't know”, since I'm not sure the register process has been triggered. To solve this "unknown" case, `Config` provides the `custom_imports` function, to make sure your module could be registered as expected.
+
+For example, we customize a optimizer:
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/config.md:478]:
```suggestion
For example, we customize an optimizer:
```

File: docs/en/advanced_tutorials/config.md [hunk 12]
```diff
@@ -1,3 +1,579 @@
...
+print(custom_optim)
+```
+
+```
+<my_module.CustomOptim object at 0x7f6983a87970>
+```
+
+### Inherit configuration files across repository
+
+It is annoying to copy a large number of configuration files when developing a new repository based on some existing repositories, `Config` support inherit configuration file from other repositories. For example,based on MMDetection, we want to develop a repository, we can use the MMDetection configuration file like this:
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/config.md:522]:
```suggestion
It is annoying to copy a large number of configuration files when developing a new repository based on some existing repositories. To address this issue, `Config` support inherit configuration files from other repositories. For example,based on MMDetection, we want to develop a repository, we can use the MMDetection configuration file like this:
```

**Review by C1rN09 — APPROVED** [review]

File: docs/en/advanced_tutorials/config.md [hunk 13]
```diff
@@ -1,3 +1,581 @@
...
+wget https://raw.githubusercontent.com/open-mmlab/mmengine/main/docs/resources/config/refer_base_var.py
+wget https://raw.githubusercontent.com/open-mmlab/mmengine/main/docs/resources/config/resnet50_delete_key.py
+wget https://raw.githubusercontent.com/open-mmlab/mmengine/main/docs/resources/config/resnet50_lr0.01.py
+wget https://raw.githubusercontent.com/open-mmlab/mmengine/main/docs/resources/config/resnet50_runtime.py
+wget https://raw.githubusercontent.com/open-mmlab/mmengine/main/docs/resources/config/resnet50.py
+wget https://raw.githubusercontent.com/open-mmlab/mmengine/main/docs/resources/config/runtime_cfg.py
+wget https://raw.githubusercontent.com/open-mmlab/mmengine/main/docs/resources/config/modify_base_var.py
+```
+
+## Read the configuration file.
```
Comment (zhouzaida) [inline docs/en/advanced_tutorials/config.md:26]:
```suggestion
## Read the configuration file
```

File: docs/en/advanced_tutorials/config.md [hunk 14]
```diff
@@ -1,3 +1,581 @@
 # Config
 
-Coming soon. Please refer to [chinese documentation](https://mmengine.readthedocs.io/zh_CN/latest/tutorials/config.html).
+MMEngine implements an abstract configuration class (`Config`) to provide a unified configuration access interface for users. `Config` supports different type of configuration file, including `python`, `json` and `yaml`, and you can choose the type according to your preference. `Config` overrides some magic method, which could help you access the data stored in `Config` just like getting values from `dict`, or getting attributes from instances. Besides, `Config` also provides an inheritance mechanism, which could help you better organize and manage the configuration files.
+
+Before starting the tutorial, let's download the configuration files needed in the tutorial(it is recommended to execute them in a temporary directory to facilitate deleting these files latter.):
```
Comment (zhouzaida) [inline docs/en/advanced_tutorials/config.md:5]:
```suggestion
Before starting the tutorial, let's download the configuration files needed in the tutorial (it is recommended to execute them in a temporary directory to facilitate deleting these files latter.):
```

File: docs/en/advanced_tutorials/config.md [hunk 15]
```diff
@@ -1,3 +1,581 @@
...
+wget https://raw.githubusercontent.com/open-mmlab/mmengine/main/docs/resources/config/resnet50.py
+wget https://raw.githubusercontent.com/open-mmlab/mmengine/main/docs/resources/config/runtime_cfg.py
+wget https://raw.githubusercontent.com/open-mmlab/mmengine/main/docs/resources/config/modify_base_var.py
+```
+
+## Read the configuration file.
+
+`Config` provides a uniform interface `Config.fromfile()` to read and parse configuration files.
+
+A legitimate configuration file should define a set of key-value pairs, and here are a few examples:
```
Comment (zhouzaida) [inline docs/en/advanced_tutorials/config.md:30]:
```suggestion
A valid configuration file should define a set of key-value pairs, and here are a few examples:
```

File: docs/en/advanced_tutorials/config.md [hunk 16]
```diff
@@ -1,3 +1,581 @@
...
+```python
+cfg = Config.fromfile('resnet50.py')
+print(cfg.optimizer)
+```
+
+```
+{'type': 'SGD', 'lr': 0.02, 'momentum': 0.9, 'weight_decay': 0.0001}
+```
+
+`_base_` is a reserved field for the configuration file. It specifies the inherited(base) files for the current file. Inheriting multiple files will get all the fields at the same time, but it requires that there is no repeated fields defined in all base files.
```
Comment (zhouzaida) [inline docs/en/advanced_tutorials/config.md:181]:
```suggestion
`_base_` is a reserved field for the configuration file. It specifies the inherited base) files for the current file. Inheriting multiple files will get all the fields at the same time, but it requires that there are no repeated fields defined in all base files.
```

File: docs/en/advanced_tutorials/config.md [hunk 17]
```diff
@@ -1,3 +1,581 @@
...
+print(cfg.a)
+```
+
+```
+{'type': 'MobileNet', 'depth': 50}
+```
+
+## Dump the configuration file
+
+The user may pass some parameters to modify some fields of the configuration file at the entry point of the training script.Therefore, we provide the `dump` method to export the changed configuration file.
```
Comment (zhouzaida) [inline docs/en/advanced_tutorials/config.md:318]:
```suggestion
The user may pass some parameters to modify some fields of the configuration file at the entry point of the training script. Therefore, we provide the `dump` method to export the changed configuration file.
```

File: docs/en/advanced_tutorials/config.md [hunk 18]
```diff
@@ -1,3 +1,581 @@
...
+````
+
+`dump_dict.py`
+
+```python
+a=1
+b=2
+```
+
+# Advanced uses
```
Comment (zhouzaida) [inline docs/en/advanced_tutorials/config.md:370]:
```suggestion
## Advanced usage
```

File: docs/en/advanced_tutorials/config.md [hunk 19]
```diff
@@ -1,3 +1,581 @@
...
+
+In this section, we'll introduce some advanced uses of the `Config`, and some tips that could make it easier for users to develop and use downstream repositories.
```
Comment (zhouzaida) [inline docs/en/advanced_tutorials/config.md:372]:
```suggestion
In this section, we'll introduce some advanced usage of the `Config`, and some tips that could make it easier for users to develop and use downstream repositories.
```

File: docs/en/advanced_tutorials/config.md [hunk 20]
```diff
@@ -1,3 +1,581 @@
...
+This can be achieved easily using predefined fields:
+
+```python
+cfg = Config.fromfile('./predefined_var.py')
+print(cfg.work_dir)
+```
+
+```
+./work_dir/predefined_var
+```
```
Comment (zhouzaida) [inline docs/en/advanced_tutorials/config.md:388]: miss some content

File: docs/en/advanced_tutorials/config.md [hunk 21]
```diff
@@ -1,3 +1,581 @@
...
+Config (path: ./example.py): {'model': {'type': 'CustomModel', 'in_channels': [1, 1, 1]}, 'optimizer': {'type': 'SGD', 'lr': 0.01}}
+```
+
+```{note}
+The standard procedure only supports modifying String, Integer, Floating Point, Boolean, None, List, and Tuple fields from the command line. For the elements of list and tuple instance, each of them must be one of the above seven types.
+```
+
+### import the custom module
+
+If we customize a module and register it into the corresponding registry, could we directly build it from the configuration file as the previous [section](#how-to-use-config) does? The answer is “I don't know”, since I'm not sure the register process has been triggered. To solve this "unknown" case, `Config` provides the `custom_imports` function, to make sure your module could be registered as expected.
```
Comment (zhouzaida) [inline docs/en/advanced_tutorials/config.md:478]:
```suggestion
If we customize a module and register it into the corresponding registry, could we directly build it from the configuration file as the previous [section](#how-to-use-config) does? The answer is "I don't know" since I'm not sure the registration process has been triggered. To solve this "unknown" case, `Config` provides the `custom_imports` function, to make sure your module could be registered as expected.
```

**Review by zhouzaida — APPROVED** [review]
